### PR TITLE
added plugin files

### PR DESCRIPTION
**Pull request by Davide-CL**

### CODE REVIEW

File: Source/FindInBlueprintExternal/Classes/FiBCommandlet.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+UCLASS(MinimalAPI)
+class UFiBCommandlet : public UCommandlet
+{
+	GENERATED_BODY()
+
+		//~ Begin UCommandlet Interface
+		virtual int32 Main(const FString& Params) override;
+	//~ End UCommandlet Interface
+
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Classes/FiBCommandlet.h]: Indenting is a bit screwy here. Should all be only one tab.

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#include "FiBCommandlet.h"
+#include "AssetRegistryModule.h"
+#include "Developer/AssetTools/Public/AssetToolsModule.h"
+#include "FindInBlueprintManager.h"
+//#include "SharedPointer.h"
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: Please write includes with paths relative to the Source folder - like "Developer/AssetTools/Public/AssetToolsModule.h", not "AssetRegistryModule.h".

Also remove the commented out SharedPointer.h include if you don't need it

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	AssetRegistryModule->Get().SearchAllAssets(true);
+
+	FString SearchValue = FParse::Token(Cmd, false);
+	TArray<FSearchResult> OutItemsFound;
+	const FStreamSearchOptions SearchOptions;
+
+	while (!SearchValue.IsEmpty()) {
+
+		if (SearchValue.Equals("-run=FiB"))
+		{
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: Inconsisted brace positioning - UE4 standard is `{` on its own line (like the if)

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		}
+
+		FStreamSearch StreamSearch(SearchValue, SearchOptions);
+
+		while (!StreamSearch.IsComplete()) FFindInBlueprintSearchManager::Get().Tick(.0);
+
+		StreamSearch.GetFilteredItems(OutItemsFound);
+
+		SearchValue = FParse::Token(Cmd, false);
+
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: You don't need a blank line between every line of code - try to group them logically

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+
+	UE_LOG(LogTemp, Warning, TEXT("Test text"));
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: Don't need this :)

You do need to log out OutItemsFound though maybe?

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    return int32();
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: we'd normally just write
```suggestion
    return 0;
```

File: Source/FindInBlueprintExternal/Private/FindInBlueprintExternal.cpp [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+void FFindInBlueprintExternalModule::ShutdownModule()
+{
+	// This function may be called during shutdown to clean up your module.  For modules that support dynamic reloading,
+	// we call this function before unloading the module.
+}
+
+#undef LOCTEXT_NAMESPACE
+	
+IMPLEMENT_MODULE(FFindInBlueprintExternalModule, FindInBlueprintExternal)
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Private/FindInBlueprintExternal.cpp]: There should be a newline at the end of the file. It helps with merging changes later

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+		if (SearchValue.Equals("-run=FiB"))
+		{
+
+			SearchValue = FParse::Token(Cmd, false);
+
+		}
+
+		FStreamSearch StreamSearch(SearchValue, SearchOptions);
+
+		while (!StreamSearch.IsComplete()) FFindInBlueprintSearchManager::Get().Tick(.0);
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: Use braces around the loop body and separate into multiple lines please

**Review by LizardThief — CHANGES_REQUESTED** [review]

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp (hunk 2 above)
Comment (Josef-Rare) [reply Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: Tooling general coding standards and examples are here https://coconutlizard.atlassian.net/wiki/spaces/CLKB/pages/823132168/MemFerret+Coding+Standards

File: Source/FindInBlueprintExternal/Classes/FiBCommandlet.h [hunk 9]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "CoreMinimal.h"
+#include "UObject/ObjectMacros.h"
+#include "Commandlets/Commandlet.h"
+#include "FiBCommandlet.generated.h"
+
+UCLASS(MinimalAPI)
+class UFiBCommandlet : public UCommandlet
```
Comment (Josef-Rare) [inline Source/FindInBlueprintExternal/Classes/FiBCommandlet.h]: General good practice to mark the class final if possible

File: Source/FindInBlueprintExternal/FindInBlueprintExternal.Build.cs [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+		PublicDependencyModuleNames.AddRange(
+			new string[]
+			{
+				"Core",
+				// ... add other public dependencies that you statically link with here ...
+			}
+			);
+			
+		
+		PrivateDependencyModuleNames.AddRange(
```
Comment (Josef-Rare) [inline Source/FindInBlueprintExternal/FindInBlueprintExternal.Build.cs:20]: Delete all the empty template AddRange calls and the 'add here' comments

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+#include "Developer/AssetTools/Public/AssetToolsModule.h"
+#include "FindInBlueprintManager.h"
+//#include "SharedPointer.h"
+
+int32 UFiBCommandlet::Main(const FString& Params)
+{
+	const TCHAR* Cmd = *Params;
+	GIsRunning = true;
+
+	FAssetRegistryModule* AssetRegistryModule = &FModuleManager::LoadModuleChecked<FAssetRegistryModule>(TEXT("AssetRegistry"));
```
Comment (Josef-Rare) [inline Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: We get a reference back from this ftn so just assign to a reference
```suggestion
	FAssetRegistryModule& AssetRegistryModule = FModuleManager::LoadModuleChecked<FAssetRegistryModule>(TEXT("AssetRegistry"));
```

Comment (Josef-Rare) [reply Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: Assigning to a pointer we lose the inherent safety of a reference, and there's no need for that here.

File: FindInBlueprintExternal.uplugin [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+	"Installed": false,
+	"EnabledByDefault": true,
+	"Modules": [
+		{
+			"Name": "FindInBlueprintExternal",
+			"Type": "Editor",
+			"LoadingPhase": "Default"
+		}
+	]
+}
```
Comment (willhinds) [inline FindInBlueprintExternal.uplugin]: Newline at the end of this file

File: Source/FindInBlueprintExternal/Classes/FiBCommandlet.h [hunk 13]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (willhinds) [inline Source/FindInBlueprintExternal/Classes/FiBCommandlet.h:3]: Do we have a Coconut Lizard Copyright notice for our Tool plugins, or do we use Epic's @Josef-CL ?

File: Source/FindInBlueprintExternal/FindInBlueprintExternal.Build.cs [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+			{
+				"Core",
+				"CoreUObject",
+				"ApplicationCore",
+				"Engine",
+				"UnrealEd",
+				"Slate",
+				"SlateCore",
+				"GameProjectGeneration",
+				"Kismet"
```
Comment (willhinds) [inline Source/FindInBlueprintExternal/FindInBlueprintExternal.Build.cs]: I think it's good to put these in alphabetical order - makes it easier to see what's there and what's not at a glance. If you haven't got it already Ori's LizardVS extension has an Alphabetiser

File: Source/FindInBlueprintExternal/Private/FindInBlueprintExternal.cpp [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+
+#include "FindInBlueprintExternal.h"
+
+#include "Modules/ModuleManager.h"
+
+#define LOCTEXT_NAMESPACE "FFindInBlueprintExternalModule"
+
+void FFindInBlueprintExternalModule::StartupModule()
+{
+	// This code will execute after your module is loaded into memory; the exact timing is specified in the .uplugin file per-module
```
Comment (willhinds) [inline Source/FindInBlueprintExternal/Private/FindInBlueprintExternal.cpp]: You can remove these comments. And probably remove both of these functions as well, unless they're pure virtual in the base class?

**Review by willhinds — CHANGES_REQUESTED** [review]

File: Source/FindInBlueprintExternal/Classes/FiBCommandlet.h (hunk 13 above)
Comment (cl-sambendell) [reply Source/FindInBlueprintExternal/Classes/FiBCommandlet.h:3]: Yeah we use `// Copyright (C) Coconut Lizard Limited. All rights reserved.`

File: Source/FindInBlueprintExternal/FindInBlueprintExternal.Build.cs [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+                "Engine",
+                "Core",
+                "CoreUObject",
+                "GameProjectGeneration",
+				"Kismet",
+				"Slate",
+                "SlateCore",
+                "UnrealEd"
+          	}
+			);
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/FindInBlueprintExternal.Build.cs]: Indenting is very broken here - mixed tab and space indenting does not work, as different people/software have different tab sizes.

**Review by LizardThief — CHANGES_REQUESTED** [review]

File: Source/FindInBlueprintExternal/FindInBlueprintExternal.Build.cs (hunk 16 above)
Comment (Josef-Rare) [reply Source/FindInBlueprintExternal/FindInBlueprintExternal.Build.cs]: We should get an editorconfig into this repo to help with indentations etc

File: Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp [hunk 17]
```diff
@@ -1,8 +1,11 @@
+// Copyright (C) Coconut Lizard Limited. All rights reserved.
+
 #include "FindInBlueprintExternal/Classes/FiBCommandlet.h"
 #include <Developer/AssetTools/Public/AssetToolsModule.h>
 #include <Editor/Kismet/Public/FindInBlueprintManager.h>
 #include <Runtime/AssetRegistry/Public/AssetRegistry/AssetRegistryModule.h>
-
+//#include "FindInBlueprintExternal/Classes/FiBResultsToJsonWriter.h"
+#include "FiBResultsToJsonWriter.h"
```
Comment (LizardThief) [inline Source/FindInBlueprintExternal/Private/FiBCommandlet.cpp]: You should only have one of these two - probably the first one by the include style?

**Review by LizardThief — CHANGES_REQUESTED** [review]

**Review by LizardThief — APPROVED** [review]

**Review by willhinds — APPROVED** [review]

**Pull request merged**